src/bit_vec.rs

```diff
@@ -34,23 +30,18 @@ impl<O: BitOrder, T: BitStore> Encode for BitSlice<T, O>
 		);
 		Compact(bits as u32).encode_to(dest);
 
-		for element in self.domain() {
+		// Iterate over chunks
+		for chunk in self.chunks(core::mem::size_of::<T>() * 8) {
+			let mut element = T::ZERO;
```

Comment:
Could this be moved outside the loop perhaps and re-used? Or will the compiler re-use `element` anyway?

Reply:
This would be wrong. As the last chunk could have less bits than the ones before, we could end up with an element having bits set that should not be set. I mean that should be fixed on the decoding side. However, I also don't see any real performance issue with having the primitive type inside the loop.